src/material/chips/chip-grid.ts

```diff
@@ -464,7 +464,12 @@ export class MatChipGrid
 
   /** Emits change event to set the model value. */
   private _propagateChanges(): void {
-    const valueToEmit = this._chips.length ? this._chips.toArray().map(chip => chip.value) : [];
+    let valueToEmit;
+    if (this.ngControl?.value) {
+      valueToEmit = this.ngControl.value;
```

Comment:
This is incorrect since it's emitting the existing value back to the form control. The chips in our example in #26358 fail, because you didn't provide a `value` to the chips. If you add `[value]="option"` to the `mat-chip-row`, it will work.

Reply:
This is the first reason, the second reason is that Angular 14 and prior `<mat-chip-list>` worked without binding the [value]="option". For me as a user, i had to debug the library in order to understand that we have to bind value to it.

Reply:
If it worked before without the `value`, it was by accident. Without the value we don't have a way of differentiating the values.

Reply:
Agree with you. I made changes by adding `displayWith` input to `MatChipRow`. Otherwise like I already said, in edit mode `MatChipEditInput` sets object in `setValue` method which sets `this.getNativeElement().textContent = value;` And since value is an object we get `[object Object]`

Reply:
@crisbeto fixed the PR and conflicts. Llease review the changes